pilot/pkg/model/gateway.go

```diff
@@ -664,3 +668,72 @@ func getTargetPortMap(serversByRouteName map[string][]*networking.Server) Gatewa
 	}
 	return pm
 }
+
+// filterMergedGatewayServers filters merged gateway servers based on strict gateway merging rules.
+// It keeps GatewayAPI servers and removes Istio Gateway servers that are not in GatewayAPI namespaces.
+// Gateways specified in gatewayNames are kept and other gateways in the same namespace are kept as well.
+func filterMergedGatewayServers(mgw *MergedGateway, gatewayNames sets.Set[string]) {
```

Comment:
is gatewayNames supposed to be only gwapi gateways? If so it would be nice to update the name of he variable.